Streetcode/Streetcode.WebApi/Utils/WebParsingUtils.cs

```diff
@@ -126,92 +127,105 @@ public async Task ParseZipFileFromWeb(string mode)
-                        string newRow = null;
+            // Grouping all rows from initial csv in order to get rid of duplicated streets
 
-                        // OSM Nominatim does not like e.g "м. Вінниця" - use "Вінниця" instead
-                        string cityStringSearchOptimized = addressEntityBase.City.Substring(addressEntityBase.City.IndexOf(" ") + 1);
+            var forParsindRows = rows
+                .Select(x => string.Join(";", x.Split(';').Take(7).ToList())).Distinct().ToList();
 
-                        string streetNameStringSearchOptimized = OptimizeStreetname(addressEntityBase.StreetName);
+            var alreadyParsedRows = allLinesFromDataCSV
+                .Select(x => string.Join(";", x.Split(';').Take(7).ToList())).Distinct().ToList();
 
-                        string addressrow = cityStringSearchOptimized + " " + streetNameStringSearchOptimized;
+            var alreadyParsedRowsToWrite = allLinesFromDataCSV
+                .Select(x => string.Join(";", x.Split(';').ToList())).Distinct().ToList();
 
-                        var res = await CallAPICoords(addressrow);
-                        if (res.Item1 == "" && res.Item2 == "")
-                        {
-                            addressrow = cityStringSearchOptimized;
-                            res = await CallAPICoords(addressrow);
-                        }
+            var remainsToParse = forParsindRows.Except(alreadyParsedRows).ToList();
+            var toBeDeleted = alreadyParsedRows.Except(forParsindRows).ToList();
 
-                        Console.WriteLine("\n" + addressrow);
-                        Console.WriteLine("Coordinates[" + res.Item1 + " " + res.Item2 + "]");
-                        addressEntityBase.Lat = res.Item1;
-                        addressEntityBase.Lon = res.Item2;
-                        newRow = addressEntityBase.Oblast + ";" + addressEntityBase.AdminRegionOld + ";" + addressEntityBase.AdminRegionNew + ";" + addressEntityBase.Gromada + ";" + addressEntityBase.City + ";" + addressEntityBase.PostIndex + ";" + addressEntityBase.StreetName + ";" + res.Item1 + ";" + res.Item2;
-                        Console.ForegroundColor = ConsoleColor.Yellow;
-                        Console.OutputEncoding = Encoding.GetEncoding(1251);
-
-                        // Console.WriteLine(newRow);
-                        Console.ForegroundColor = ConsoleColor.White;
-                        await File.AppendAllTextAsync(csvPath, newRow + "\n", Encoding.GetEncoding(1251));
-                    }
-                }
-            }
-            catch (Exception ex)
+            Console.WriteLine("Remains to parse: " + remainsToParse.Count());
+            Console.WriteLine("To be deleted: " + toBeDeleted.Count());
+
+            foreach (var row in toBeDeleted)
```

Comment:
This foreach call may be converted to an `.Aggregate()` invocation.

Reply:
i dont know how, maybe in next updates

Reply:
Ok, it is not an urgent remark.